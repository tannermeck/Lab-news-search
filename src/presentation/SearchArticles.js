import React from 'react';

export const SearchArticles = ({onSubmit, query, handleChange}) => {
    return (
        <>
            <form onSubmit={onSubmit}>
                <label htmlFor="search">Find Articles</label>
                <input id="search" name="search" value={query} onChange={handleChange}/>
                <button aria-label="button">Search Articles!</button>
            </form>
        </>
    )
}