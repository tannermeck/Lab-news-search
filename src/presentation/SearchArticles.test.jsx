import React from 'react';
import {render} from '@testing-library/react';
import { SearchArticles } from './SearchArticles';

describe('should imitate a search query', () => {
    it('should return a list of articles related to the search', () => {
        const { asFragment } = render(
            <SearchArticles onSubmit='health' query='health' handleChange='health'/>
        )
        expect(asFragment()).toMatchSnapshot()
    })
})