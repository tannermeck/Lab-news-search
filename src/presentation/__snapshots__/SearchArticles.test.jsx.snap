// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should imitate a search query should return a list of articles related to the search 1`] = `
<DocumentFragment>
  <form>
    <label
      for="search"
    >
      Find Articles
    </label>
    <input
      id="search"
      name="search"
      value="health"
    />
    <button
      aria-label="button"
    >
      Search Articles!
    </button>
  </form>
</DocumentFragment>
`;
